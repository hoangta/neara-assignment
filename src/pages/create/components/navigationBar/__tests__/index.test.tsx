import React from "react";
import { render, fireEvent } from '@testing-library/react';
import NavigationBar from "..";

describe('Pages/Create/Components/NavigationBar', () => {
    const onBackSpy = jest.fn();
    const onCreateSpy = jest.fn();

    it('should render correctly with Add button enabled', () => {
        const { asFragment, getByTestId } = render(<NavigationBar onBack={onBackSpy} onCreate={onCreateSpy} canCreate />);
        const addButton = getByTestId('add_button');
        expect(addButton).not.toBeDisabled();
        expect(asFragment()).toMatchSnapshot();
    })

    it('should render correctly with Add button disabled', () => {
        const { asFragment, getByTestId } = render(<NavigationBar onBack={onBackSpy} onCreate={onCreateSpy} canCreate={false} />);
        const addButton = getByTestId('add_button');
        expect(addButton).toBeDisabled();
        expect(asFragment()).toMatchSnapshot();
    })

    it('should trigger onBack function when click Back button', () => {
        const { getByTestId } = render(<NavigationBar onBack={onBackSpy} onCreate={onCreateSpy} canCreate />);
        const backButton = getByTestId('back_button');
        fireEvent.click(backButton)
        expect(onBackSpy).toHaveBeenCalledTimes(1);
    })

    it('should trigger onCreate function when click Add button', () => {
        const { getByTestId } = render(<NavigationBar onBack={onBackSpy} onCreate={onCreateSpy} canCreate />);
        const addButton = getByTestId('add_button');
        fireEvent.click(addButton)
        expect(onCreateSpy).toHaveBeenCalledTimes(1);
    })
})