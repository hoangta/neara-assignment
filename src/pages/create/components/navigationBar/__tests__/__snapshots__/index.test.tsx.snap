// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Create/Components/NavigationBar should render correctly with Add button disabled 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="button"
      data-testid="back_button"
    >
      Back
    </button>
    <div
      class="title"
    >
      New token
    </div>
    <button
      class="button"
      data-testid="add_button"
      disabled=""
    >
      Add
    </button>
  </div>
</DocumentFragment>
`;

exports[`Pages/Create/Components/NavigationBar should render correctly with Add button enabled 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="button"
      data-testid="back_button"
    >
      Back
    </button>
    <div
      class="title"
    >
      New token
    </div>
    <button
      class="button"
      data-testid="add_button"
    >
      Add
    </button>
  </div>
</DocumentFragment>
`;
